cpp/oneapi/dal/algo/basic_statistics/backend/gpu/compute_kernel_dense_impl_dpc.cpp

```diff
@@ -193,8 +192,7 @@ struct singlepass_processor_kernel {
         (DefferedFin && check_mask_flag(bs_list::varc | bs_list::stdev | bs_list::vart, params));
     std::conditional_t<output_sum2cent, Float*, empty> sum2cent_ptr;
 
-    constexpr static inline bool output_mean =
```

Comment:
Why can `!DeferredFin` be removed?

Reply:
I've removed `!DeferredFin` because I've changed logic for merging data from different gpus. In this implementation I do use local means not only for single gpu, where its final result, also do it for distributed as a subproduct for sum2cent computation.
The idea of optimizations was based on simplifying merging function. I changed logic from comm.allgather and loop(column count x rank_count) which compute a lot of statistics on data after allgather, to comm.allreduce and small loop(column count) for computing data which based only on merged data. Also, bigger loop computed sum+means(as a current loop's sum divided by current mean's scale)+sum2cent, new one just sum2cent+means(as a allreduced sum + number of rows)